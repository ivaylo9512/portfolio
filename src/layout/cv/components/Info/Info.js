import './Info.scss';
import Data from './Data/Data';
import ProfileImage from './ProfileImage/ProfileImage';

const Info = () => {
    return(
        <div className='profile-info'>
            <ProfileImage />
            <Data />
        </div>
    )
}

export default Info