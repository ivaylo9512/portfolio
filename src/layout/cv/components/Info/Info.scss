.profile-info{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 1% 0;
    width: 100%;

    .image-container{
        justify-content: center;
        align-items: center;
        display: flex;
        width: 70%;
        
        img{
            width: 100%;
        }
    }

    .data{
        flex-direction: row;
        width: 70%;
        font-size: 2.9vw;
        justify-content: center;
        font-family: Abril;
        display: flex;
        font-weight: bold;
        color: black;
        box-sizing: border-box;
        flex-wrap: wrap;
  
        a:hover,
        a:focus{
            transform: scale(1.1);
        }

        > div{
            width: 50%;
            display: flex;
            align-items: center;
            justify-content: flex-start;
            margin-bottom: 4%;
        }

        > div:nth-child(2n + 3){
            justify-content: flex-end;

            .info-icon{
                margin-left: 2%;
                margin-right: 0;
            }
        }

        .name{
            width: 100%;
            justify-content: center;

            .info-icon{
                width: 12%;
            }
        } 

        .info-name{
            width: 93%;
        }

        .info-icon{
            rect{
                filter: url(#info-shadow);
                fill: white;
            }

            width: 20%;
            margin-right: 2%;
        }
    }

    @media only screen and (min-width: 700px) and (orientation: landscape) {
        justify-content: flex-end;
        flex-direction: row;

        .image-container{
            width: 20%;
            justify-content: flex-end;
            align-items: center;

            img{
                margin-right: 4%;
            }
        }

        .data{
            font-size: 0.85vw;
            padding-left: 3%;
            width: 42%;

            .info-icon{
                width: 14%;
                margin-right: 4%;
            }

            > div {
                margin-bottom: 3%;
            }

            > div:nth-child(2n + 3){
                flex-direction: row-reverse;

                .info-icon{
                    margin-right: 4%;
                    margin-left: 0;
                }
            }

            .name{
                justify-content: left;

                .info-icon{
                    margin-right: 2%;
                    width: 8%;
                }
            }

            .info-name{
                width: 61%;
            }

            .phone{
                margin-bottom: 2%;
            }
        }
    }
    @media only screen and (min-width : 896px) and (max-width : 1100px) {
        .data{
            font-size: 1.8vw;
        }
    } 

    @media only screen and (max-width : 896px) and (orientation : landscape) { 
        .data{
            font-size: 1.35vw;
        }
    }
}