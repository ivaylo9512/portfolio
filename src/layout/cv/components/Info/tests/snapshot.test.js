import Info from '../Info';
import { shallow } from 'enzyme';
import * as Redux from 'react-redux';

describe('Info snapshot tests', () => {
    let selectorSpy = jest.fn();

    beforeEach(() => {
        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (state) => {
        selectorSpy.mockReturnValue(state);
        
        return shallow(
            <Info />
        )
    }

    it('should match snapshot with isProjects to true', () => {
        const wrapper = createWrapper(true);
     
        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot with isProjects to false', () => {
        const wrapper = createWrapper(false);
     
        expect(wrapper).toMatchSnapshot();
    })
});