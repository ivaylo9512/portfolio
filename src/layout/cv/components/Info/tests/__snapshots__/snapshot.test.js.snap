// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info snapshot tests should match snapshot with isProjects to false 1`] = `
<div
  className="profile-info"
>
  <div
    className="image-container"
  >
    <img
      src="profile-image.png"
    />
  </div>
  <div
    className="data"
  >
    <div
      className="phone"
    >
      <Phone />
      0886892842
    </div>
    <div>
      <GithubSquare />
      <a
        data-testid="github-link"
        href="https://github.com/ivaylo9512"
      >
        Github
      </a>
    </div>
  </div>
</div>
`;

exports[`Info snapshot tests should match snapshot with isProjects to true 1`] = `
<div
  className="profile-info"
>
  <div
    className="image-container"
  >
    <img
      src="profile-image.png"
    />
  </div>
  <div
    className="data"
  >
    <div
      className="phone"
    >
      <Phone />
      0886892842
    </div>
    <div>
      <GithubSquare />
      <a
        data-testid="github-link"
        href="https://github.com/ivaylo9512"
        tabIndex="-1"
      >
        Github
      </a>
    </div>
  </div>
</div>
`;
